examples/flava/data/datapipes.py

```diff
@@ -0,0 +1,428 @@
+
+
+class DataCollatorForWholeWordMaskRetainingBatch(DataCollatorForWholeWordMask):
+    def torch_call(
+        self, examples: List[Union[List[int], Any, Dict[str, Any]]]
+    ) -> Dict[str, Any]:
+        masked_batch = super().torch_call(examples)
+        examples = torch_default_data_collator(examples)
+        examples["input_ids"] = masked_batch["input_ids"]
+        examples["labels"] = masked_batch["labels"]
+        return examples
+
+class TextDataPipeModule(LightningDataModule):
+    def __init__(
+        self,
+        train_infos: List[HFDatasetInfo],
+        text_columns: List[str],
+        val_infos: Optional[List[HFDatasetInfo]] = None,
+        tokenizer: Optional[Callable] = None,
+        max_length: int = 512,
+        batch_size: int = 32,
+        num_workers: int = 4,
+        allow_uneven_batches: bool = False,
+        **kwargs: Any,
+    ):
+        super().__init__()
+        self.train_dataset_infos = train_infos
+        self.text_columns = text_columns
+        self.val_dataset_infos = val_infos
+        if self.val_dataset_infos is None:
+            self.val_dataset_infos = train_infos
+        self.tokenizer = tokenizer
+        self.max_length = max_length
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.allow_uneven_batches = allow_uneven_batches
+
+    def setup(self, stage=None):
+        if self.tokenizer is None:
+            self.tokenizer = BertTokenizer.from_pretrained(TEXT_DEFAULT_TOKENIZER)
+        transform = partial(
+            encode_text_batch,
+            tokenizer=self.tokenizer,
+            padding="max_length",
+            max_length=self.max_length,
+            truncation=True,
+            return_tensors="pt",
+            return_special_tokens_mask=True,
+            text_columns=self.text_columns,
+            return_batch=True,
+        )
+
+        # previous HF dataset
+        self.train_dataset = build_datasets_from_info(
+            self.train_dataset_infos, split="train"
+        )
+        self.train_dataset.set_transform(transform)
+        self.val_dataset = build_datasets_from_info(
+            self.val_dataset_infos, split="validation"
+        )
+        self.val_dataset.set_transform(transform)
+
+        self.train_datapipe_source = IterableWrapper(self.train_dataset)
+        self.val_datapipe_source = IterableWrapper(self.val_dataset)
+
+    def train_datapipe(self):
+        return self._build_datapipe(datapipe=self.train_datapipe_source,collate_fn=self._build_collator)
+
+    def val_datapipe(self):
+        return self._build_datapipe(datapipe=self.val_datapipe_source, collate_fn=self._build_collator, shuffle=False)
+
+    def _build_datapipe(self, datapipe: IterDataPipe,  collate_fn : Any, drop_last: bool = False, shuffle: bool = True):
+        output_datapipe = datapipe.batch(batch_size=self.batch_size, drop_last=drop_last).collate(collate_fn)
+        if shuffle:
+            return output_datapipe.shuffle()
```

Comment:
`sharding_filter` should be placed after `shuffle` here. And, you can always add `shuffle` in the pipeline. If you want to disable shuffling explicitly, you can either:
- `shuffle_dp = shuffle_dp.set_shuffle(False)`
- `DataLoader(datapipe, shuffle=False)`